app/single_resource/views.py

```diff
@@ -45,24 +59,108 @@ def create():
 def edit(resource_id):
     """Edit a resource."""
     resource = Resource.query.get(resource_id)
-    field_names = Resource.__table__.columns.keys()
+    if resource is None:
+        abort(404)
+    resource_field_names = Resource.__table__.columns.keys()
+    descriptors = Descriptor.query.all()
+    for descriptor in descriptors:
+        # fields for option descriptors
+        if (descriptor.values):
+            choices = [(str(i), v) for i, v in enumerate(descriptor.values)]
+            default = None
+            option_association = OptionAssociation.query.filter_by(
+                resource_id=resource_id,
+                descriptor_id=descriptor.id
+            ).first()
+            if (option_association is not None):
```

Comment:
If an option_association between the current resource and descriptor exists, (see line 71), it is input into the form as the default choice